proto/resource_ops.proto

```diff
@@ -0,0 +1,80 @@
+syntax = "proto3";
+
+package rs.tokio.console.resource_ops;
+
+import "google/protobuf/timestamp/timestamp.proto";
+import "google/protobuf/duration.proto";
+import "common.proto";
+
+
+// An AsyncOp state update.
+message ResourceOpUpdate {
+    repeated ResourceOp new_resource_ops = 1;
+}
+
+message ResourceOp {
+    // The resources's ID.
+    //
+    // This uniquely identifies this op across all *currently live*
+    // ones.
+    common.MetaId id = 1;
+    // The numeric ID of the op's `Metadata`.
+    //
+    // This identifies the `Metadata` that describes the `tracing` span
+    // corresponding to this op. The metadata for this ID will have been sent
+    // in a prior `RegisterMetadata` message.
+    common.MetaId metadata = 2;
+    // The resources's ID.
+    common.SpanId resource_id = 3;
+    OpType op_type = 4;
+    // the name of this op (e.g. poll_elapsed, new_timeout, reset, etc.)
+    string name = 5;
+}
+
+
+message OpType {
+    oneof op_type {
+        // An op that reflects changes to the state of the resource.
+        StateUpdate state_update = 1;
+        // An op that reflects the invocation of a `poll` method
+        Poll poll = 2;
+    };
+
+    message StateUpdate {
+        repeated AttributeUpdate updates = 1;
+
+        message AttributeUpdate {
+            string name = 1;
+            oneof update {
+                Numeric numeric = 2;
+                string text = 3;
+            };
+
+            message Numeric {
+                uint64 val = 1;
+                Op op = 2;
+                string unit = 3;
+
+                enum Op {
+                    ADD = 0;
+                    OVR = 1;
+                    SUB = 2;
+                }
+            }
```

Comment:
if i'm understanding this correctly, it appears that this represents a kind of diff-based way of updating a numeric value? when a numeric value associated with a resource changes, we send an add or sub message indicating the difference from the previous value?

i think this approach to tracking numeric values seems a bit unnecessarily complex and inconsistent with the rest of the wire proto. elsewhere, for pretty much every numeric value in the wire protocol (e.g. number of polls/wakers/etc), we just always send the current value, rather than sending a diff. is there a reason we can't do that here as well? it should simplify the implementation significantly.

Reply:
Alright, yes that makes total sense. I will just emit the updated current value.

Reply:
This brings another questions. Currently the aggregator actually performs update on the these `state` values and sends the updated ones as part of the resource stats (`Stats::attributes`) That being said, should we just get rid of this resource ops message and not send it and rely on the resource stats reflecting the current values of the resource attribute?